pkg/R/utils.R

```diff
@@ -3,56 +3,68 @@
 # Given a JList<T>, returns an R list containing the same elements, the number
 # of which is optionally upper bounded by `logicalUpperBound` (by default,
 # return all elements).  Takes care of deserializations and type conversions.
-convertJListToRList <- function(jList, flatten, logicalUpperBound = NULL, serialized = TRUE) {
+convertJListToRList <- function(jList, flatten, logicalUpperBound = NULL,
+  serialized = "byte", colNames = list()) {
   arrSize <- callJMethod(jList, "size")
 
   # Unserialized datasets (such as an RDD directly generated by textFile()):
   # each partition is not dense-packed into one Array[Byte], and `arrSize`
   # here corresponds to number of logical elements. Thus we can prune here.
-  if (!serialized && !is.null(logicalUpperBound)) {
+  if (serialized == "string" && !is.null(logicalUpperBound)) {  
     arrSize <- min(arrSize, logicalUpperBound)
   }
 
   results <- if (arrSize > 0) {
     lapply(0:(arrSize - 1),
-           function(index) {
-             obj <- callJMethod(jList, "get", as.integer(index))
+          function(index) {
+            obj <- callJMethod(jList, "get", as.integer(index))
 
-             # Assume it is either an R object or a Java obj ref.
-             if (inherits(obj, "jobj")) {
-               if (isInstanceOf(obj, "scala.Tuple2")) {
-                 # JavaPairRDD[Array[Byte], Array[Byte]].
+            # Assume it is either an R object or a Java obj ref.
+              if (inherits(obj, "jobj")) {
```

Comment:
Updating the formatting for the whole function since it was all messed up. It'd be great if @hlin09 and @shivaram could give it a second look.